### ISSUE TITLE
Trusted book provider "audiobook" link in carousel on mobile has text overlap with external icon

### ISSUE THREAD

**Issue by RayBB** [internetarchive/openlibrary#8899]
### Problem

Hard to explain but see screenshot of "audiobook" button

#### Evidence / Screenshot
![Screen Shot 2024-03-12 at 17 44 14](https://github.com/internetarchive/openlibrary/assets/921217/b13f7908-a384-4a14-b22a-2f8c21299652)

#### Relevant URL(s)
https://openlibrary.org/works/OL37473900W/Short_Story_Collection_102

### Reproducing the bug

* Expected behavior: No overlap
* Actual behavior: Overlap

### Context

- Browser (Chrome, Safari, Firefox, etc): FF
- OS (Windows, Mac, etc): Android
- Logged in (Y/N): Y
- Environment (prod, dev, local): prod

#### Proposal & constraints

If you want to work on this please confirm that you will read the feedback and progress in the closed PR https://github.com/internetarchive/openlibrary/pull/9066 and incorporate it into the PR you open.

#### Stakeholders

@cdrini

**Comment by PremBharwani** [comment]
Hi, I would like to work on this. The workflow on top of my head is(I'll throw in more details soon): 
- reproduce the issue
- identify the cause & fix that
- check if the issue doesn't occur on different device types & resolutions
- open a PR as per the guidelines

**Comment by RayBB** [comment]
@PremBharwani I assigned you. Let me know if I can help at all

**Comment by RayBB** [comment]
This issue is open for someone to work on again. You may take inspiration from #8913 which started but still needs work.

**Comment by Realmbird** [comment]
If I can I work on this issue. Please assign me

**Comment by camicacace** [comment]
Hi, is this issue still open?

**Comment by RayBB** [comment]
@camicacace yes and there is a PR open awaiting review. If you want to find an issue please look at the good first issues with no assignee here: https://github.com/internetarchive/openlibrary/issues?q=is%3Aissue+is%3Aopen+-linked%3Apr+label%3A%22Good+First+Issue%22+no%3Aassignee

**Comment by LouisSim** [comment]
I can work on this issue. Could you please assign me?

**Comment by LouisSim** [comment]
@RayBB https://github.com/internetarchive/openlibrary/pull/10112 PR has been submitted for this issue, let me know if there is anything I need to change or fix
